op-defender/psp_executor/defender.go

```
@@ -471,47 +510,64 @@ func getLatestPSP(pspData []PSP, nonce uint64) (PSP, error) {
 	if err != nil {
 		log.Error("failed to check the pause status of the SuperChainConfig", "error", err, "superchainconfig_address", d.superChainConfigAddress)
 		return common.Hash{}, err
 	}
-	if pause_before_transaction {
 
+	if pause_before_transaction {
 		return common.Hash{}, errors.New("the SuperChainConfig is already paused")
+	}
+	d.log.Info("[Before Transaction] status of the pause()", "pause", pause_before_transaction)
+	d.log.Info("Current parameters", "SuperchainConfigAddress", d.superChainConfigAddress, "safe_address", d.safeAddress, "chainID", d.chainID)
 
+	// 2. Simulate the transaction to check if it will succeed before sending it onchain if blocknumber = nil this simulate at the last block
+	simulation, err := SimulateTransaction(ctx, d.l1Client, nil, d.senderAddress, safe_address, calldata)
+	if err != nil {
+		d.log.Warn("🛑 Simulated transaction failed 🛑", "from", d.senderAddress, "to", safe_address, "error", err.Error())
+		return common.Hash{}, fmt.Errorf("failed to simulate transaction: %v", err)
 	}
-	log.Info("[Before Transaction] status of the pause()", "pause", pause_before_transaction)
-	log.Info("Current parameters", "SuperchainConfigAddress", d.superChainConfigAddress, "safe_address", d.safeAddress, "chainID", d.chainID)
 
+	d.log.Info("✅ Simulated transaction succeed ✅", "from", d.senderAddress, "to", safe_address, "simulation", hex.EncodeToString(simulation))
+	// 3. As the simulation is correct we execute the transaction onchain.
 	txHash, err := sendTransaction(d.l1Client, d.chainID, d.privatekey, safe_address, big.NewInt(0), calldata) // Send the transaction to the chain with 0 wei.
 	if err != nil {
 		return common.Hash{}, err
 	}
-	log.Info("Transaction sent!", "TxHash", txHash)
+	d.log.Info("Transaction sent!", "TxHash", txHash)
 
+	// 4. Check the status of the SuperChainConfig after the transaction is set to pause.
 	pause_after_transaction, err := d.checkPauseStatus(ctx)
 	if !pause_after_transaction {
```

Not Checking error, but using returned result to check if it should raise an error.